tests/simulations/test_simulation_runner.py

```diff
@@ -107,9 +107,12 @@ def test_simulation_runner_send_messages(
         False
     )
 
-    assert len(simulation_runner.network.global_view.justified_messages) == len(validator_set)
+    if protocol == BlockchainProtocol:
+        assert len(simulation_runner.network.global_view.justified_messages) == 0
+    if protocol == BinaryProtocol:
+        assert len(simulation_runner.network.global_view.justified_messages) == len(validator_set)
 
     for i in range(10):
         simulation_runner.step()
-        assert len(simulation_runner.network.global_view.justified_messages) == \
-            (i + 1) * messages_generated_per_round + len(validator_set)
+        assert len(simulation_runner.network.global_view.justified_messages) <= \
```

Comment:
capture the initial length of the `justified_messages`, plug it into where `len(validator_set)` is below, and use `==`